hermes/file_gen_test.go

```diff
@@ -25,18 +28,18 @@ type FileNameTestUnit struct {
 	want     string
 }
 
-var test_unit_table [5]FileNameTestUnit
+var testUnitTable [5]FileNameTestUnit
 
-// Function completes the table of HermesFileTestUnit by iterating over its entries
+// completeTestUnitTable completes the table of HermesFileTestUnit by iterating over its entries
 func completeTestUnitTable() {
 	for i := 0; i < 5; i++ {
-		test_unit_table[i].id = i*10 + 1
-		test_unit_table[i].checksum = "abba"
-		test_unit_table[i].want = fmt.Sprintf("Hermes_%02d_%v", test_unit_table[i].id, test_unit_table[i].checksum)
+		testUnitTable[i].id = i*10 + 1
+		testUnitTable[i].checksum = "abba"
+		testUnitTable[i].want = fmt.Sprintf("Hermes_%02d_%v", testUnitTable[i].id, testUnitTable[i].checksum)
 	}
 }
 
-// Test function for the generateFileChecksum function in file_gen.go
+// TestChecksum tests the generateFileChecksum() method.
```

Comment:
This is for line 49:
The idiomatic error message would be

`t.Errorf("generateFileChecksum() = %v; want %v")`

Also note that unit tests should exercise the public API of the package. In rare cases it makes sense to test some non-exported function/method. In this case it's not necessary.

Reply:
For now the file contents are just a simple string but I will change it soon and we will use read.io and a pseudo random byte generator, therefore I wanted to keep the TestChecksum function in the test file. Is this alright?

Reply:
I would implement a single test that creates HermesFile instances and checks that the filename, contents and checksum have the expected values.
The tests should exercise the system to the fullest extent feasible. If the test only exercise individual functions, then there's no guarantee that the package-level behaviour is correct.
For example, this test passes, but then someone deletes the line that calls this function. The result will be an empty checksum and no test failures.

Reply:
I would like to keep these test functions to test the individual components here on their own since the rest of Hermes's logic depends on the correctness of these components but will add the function that tests everything together once the read.io integration is complete!